modules/core/misc/java/src/java/core+MatOfByte.java

```diff
@@ -48,6 +53,14 @@ public void fromArray(byte...a) {
         put(0, 0, a); //TODO: check ret val!
     }
 
+    public void fromArray(int offset, int length, byte...a) {
+        if(a==null || a.length==0 || a.length - (length + offset) < 0)
+            return;
```

Comment:
> a.length - (length + offset) < 0

I believe we should have here the OutOfRange exception.

P.S. Add "offset >= 0" check.

Reply:
Hi, I agree with offset >= 0 (or "offset < 0" to return) but the original function doesn't throw any exception. Should I throw the exception or skip the check to let it naturally fail with OutOfRange exception?

Reply:
`offset < 0` should throw OutOfRangeException too instead of return.

There is missing exception throwing on "a.length - (length + offset) < 0" condition in the last commit.

Reply:
Should we actually have something like `throw new OutOfRangeException()`? I am a bit confused, because in the original function a.length=0 returns nothing (no exception is thrown). Conceptially, `offset < 0` is similar to a `a=null` or `a.length=0`. First should have thrown NullPointerException and the second should have thrown IllegalArgumentException. Instead the original function returns. I did not want to change that design. What do you think?

Reply:
Negative "offset" or "length" doesn't make any sense - it is programming error in general. Good library should point on this by raising an error message (detect errors early).
Bad libraries silently "eat" such errors and provide wrong results in the end of complex processing pipeline - hours of hard debugging are guaranteed.

I mean this:
```
if (offset < 0)
    throw new OutOfRangeException(...);
if (length < 0)
    throw new OutOfRangeException(...);
if (a==null || a.length==0)
    return; // keep original behavior, but NPE is preferable for a==null
if (a.length - (length + offset) < 0)
    throw new OutOfRangeException(...);
```